server/src/main/scala/com/stratio/crossdata/server/CrossdataHttpServer.scala

```diff
@@ -45,41 +46,43 @@ class CrossdataHttpServer(config: Config, serverActor: ActorRef, implicit val sy
   lazy val logger = Logger.getLogger(classOf[CrossdataHttpServer])
   lazy val mediator = DistributedPubSubExtension(system).mediator
 
+  type SessionDirective[Session] = Directive[Tuple1[Session]]
+
   def route =
-    path("upload") {
-      entity(as[Multipart.FormData]) {
-        formData =>
-          // collect all parts of the multipart as it arrives into a map
-          var path = ""
-          val allPartsF: Future[Map[String, Any]] = formData.parts.mapAsync[(String, Any)](1) {
-
-            case part: BodyPart if part.name == "fileChunk" =>
-              // stream into a file as the chunks of it arrives and return a future file to where it got stored
-              val file = new java.io.File(s"/tmp/${part.filename.getOrElse("uploadFile")}")
-              path = file.getAbsolutePath
-              logger.info("Uploading file...")
-              // TODO map is not used
-              part.entity.dataBytes.runWith(FileIO.toFile(file)).map(_ => part.name -> file)
-
-          }.runFold(Map.empty[String, Any])((map, tuple) => map + tuple)
-
-          val done = allPartsF.onSuccess {
-            case _ => logger.info("Recieved file")
-          }
+    path("upload" / JavaUUID) { sessionUUID =>
+      entity(as[Multipart.FormData]) { formData =>
+        // collect all parts of the multipart as it arrives into a map
+        var path = ""
+        val allPartsF: Future[Map[String, Any]] = formData.parts.mapAsync[(String, Any)](1) {
+
+          case part: BodyPart if part.name == "fileChunk" =>
+            // stream into a file as the chunks of it arrives and return a future file to where it got stored
+            val file = new java.io.File(s"/tmp/${part.filename.getOrElse("uploadFile")}")
+            path = file.getAbsolutePath
+            logger.info("Uploading file...")
+            // TODO map is not used
+            part.entity.dataBytes.runWith(FileIO.toFile(file)).map(_ => part.name -> file)
 
-          // when processing have finished create a response for the user
-          onSuccess(allPartsF) { allParts =>
-            complete {
-              val hdfsConfig = XDContext.xdConfig.getConfig("hdfs")
-              //Send a broadcast message to all servers
-              val hdfsPath = writeJarToHdfs(hdfsConfig, path)
-              mediator ! Publish(AddJarTopic, CommandEnvelope(AddJARCommand(hdfsPath, hdfsConfig = Option(hdfsConfig)), new Session(UUID.randomUUID(), serverActor)))
-              hdfsPath
-            }
+        }.runFold(Map.empty[String, Any])((map, tuple) => map + tuple)
+
+        val done = allPartsF.onSuccess {
+          case _ => logger.info("Recieved file")
```

Comment:
It looks like value done is not used in any place. In addition, maybe we could take advantage of the "onSuccess(allPartsF)" in order to write in the log and remove this part